### PR TITLE
Migrate from satori/go.uuid to gofrs/uuid

### PR DESCRIPTION
**Pull request by oksanabaza**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Replace satori/go.uuid with gofrs/uuid to remove a CVE.
#### Which issue(s) this PR fixes:

Fixes # High CVEs

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by oksanabaza** [comment]
@feiskyer and @x13n could you please review this PR?

File: cluster-autoscaler/go.mod [hunk 1]
```diff
@@ -263,3 +262,5 @@ replace k8s.io/dynamic-resource-allocation => k8s.io/dynamic-resource-allocation
 replace k8s.io/kms => k8s.io/kms v0.29.0-alpha.3
 
 replace k8s.io/endpointslice => k8s.io/endpointslice v0.29.0-alpha.3
+
+replace github.com/satori/go.uuid => github.com/gofrs/uuid/v5 v5.0.0
```
Comment (alculquicondor) [inline cluster-autoscaler/go.mod:266]: if you are removing the usage of `/satori`, why do we need the replace?

File: cluster-autoscaler/cloudprovider/alicloud/alibaba-cloud-sdk-go/sdk/utils/utils.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ import (
 	"encoding/hex"
 	"encoding/json"
 	"fmt"
-	"github.com/satori/go.uuid"
+	"github.com/gofrs/uuid"
```
Comment (alculquicondor) [inline cluster-autoscaler/cloudprovider/alicloud/alibaba-cloud-sdk-go/sdk/utils/utils.go:25]: should we be using `github.com/google/uuid` instead? For consistency with kubernetes/kubernetes.

Are there any advantages to either?

File: cluster-autoscaler/go.mod (hunk 1 above)
Comment (oksanabaza) [reply cluster-autoscaler/go.mod]: satori replace is deleted, thank you

File: cluster-autoscaler/cloudprovider/alicloud/alibaba-cloud-sdk-go/sdk/utils/utils.go (hunk 2 above)
Comment (oksanabaza) [reply cluster-autoscaler/cloudprovider/alicloud/alibaba-cloud-sdk-go/sdk/utils/utils.go]: make sense as we already have `github.com/google/uuid` in dependency, I've replaced `gofrs/uuid` with `github.com/google/uuid` using the latest version

File: cluster-autoscaler/go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module k8s.io/autoscaler/cluster-autoscaler
 
-go 1.21
+go 1.21.6
```
Comment (towca) [inline cluster-autoscaler/go.mod:3]: Why is the required Go version bumped? It doesn't seem like github.com/google/uuid v1.6.0 requires anything higher than go1.21.

Comment (oksanabaza) [reply cluster-autoscaler/go.mod]: @towca thanks, I returned the previous version 1.21

**Review by alculquicondor — COMMENTED** [review]
I forgot to click "Submit review" :(

**Comment by ddl-risong-na** [comment]
@oksanabaza any plans to apply the same change for previous versions, 1.27, 1.28 and 1.29? The CVEs still exist for those versions. Thank you!

**Pull request merged**